web/auth/oauth.go

```diff
@@ -545,6 +545,15 @@ func (a *AuthorizeHTTPHandler) authorizeSharingForm(c echo.Context) error {
 		if err := s.AddShortcut(instance, params.state); err != nil {
 			return err
 		}
+
+		if s.Drive && !isMemberReady(s, params.state) {
```

Comment:
I think the second condition, `!isMemberReady(s, params.state)` should be nested in the `s.Drive` conditional block, after we check that `params.state` is not empty as it does not make much sense to check if the member is ready based on an empty state.